### ISSUE TITLE
Fedora package should not call unversioned `python`

### ISSUE THREAD

**Issue by cbm755** [cbm755/octsympy#879]
@mtmiller do you maintain the Debian package?  Is there is a similar issue to this?

[Fedora packaging guidelines](https://fedoraproject.org/wiki/Packaging:Python#Multiple_Python_Runtimes) say:

> However, packages in Fedora MUST NOT depend on
> where /usr/bin/python happens to point but instead
> MUST call the proper executable for the needed python
> major version directly, either /usr/bin/python2 or
> /usr/bin/python3 as appropriate.

We do call `python` by default.  The Fedora package also depends explicitly on `python2-sympy`.  So we will break when Fedora switches the default Python to 3.

Some options:

  * Make the Fedora package depend on both `python2-sympy`
    and `python3-sympy`.  Then it will work whether `python` is 2
    or 3.  But its useless dependency bloat.

  * We could `sed` the `symbolic` pkg code (during Fedora package
    build) to call `python3` by default (when `$PYTHON` is unset).
    Then the package should depend on `python3-sympy`.

I'm leaning towards the second option.

**Comment by mtmiller** [comment]
Debian's Python policy is a little more forgiving.

> Python scripts _should_ specify the Debian Python interpreter, to ensure that the Debian Python installation is used and all dependencies on additional Python modules are met.
> Scripts requiring the default Python 2 version may instead specify the interpreter /usr/bin/python.

Emphasis mine.

I interpret that as everything being fine as is. The Debian package depends on `python-sympy` which is the Python 2 version.

What, if anything, could be done here to make it easier for a package maintainer to choose an appropriate default for the distribution? Is there a way that the `DEFAULTPYTHON` variable could be made into a documented constant? Call it out and document it as something that a distributor / vendor might want to change?

**Comment by cbm755** [comment]
I suppose exposing `DEFAULTPYTHON` as an environment variable is a bit heavy.  Do you think it would help to have a `private/defaultpython.m` file, with some discussion/docs in it?  `sympref` could then call that file.

The `symbolic-win-py-bundle-x.y.z.zip` currently uses `sed` to replace `DEFAULTPYTHON = python` with `DEFAULTPYTHON = octpy.exe`; I will probably do this for Fedora.

for reference: https://www.python.org/dev/peps/pep-0394

**Comment by mtmiller** [comment]
Yeah I guess as a Debian packager I'd rather see a documented way for me to hardcode the appropriate python interpreter into the package when building it, just so I can be assured that that is the one place and it is a supported and expected thing to be doing.

`inst/private/defaultpython.m` looks good to me, basically just something like

```
function s = defaultpython ()
  persistent python = 'python';
  s = python;
end
```

?

**Comment by mtmiller** [comment]
> Is the persistent thing to avoid thrashing memory?

It's just the idiomatic way to define a function that returns a constant value, not necessary.

**Comment by mtmiller** [comment]
This issue has inspired me to switch the Debian octave-symbolic package to Python 3 when your next release drops.
